R/tpch-queries.R

```diff
@@ -695,7 +695,7 @@ tpc_h_queries[[21]] <- function(input_func, collect_func = dplyr::collect, con =
     inner_join(input_func("orders"), by = c("l_orderkey" = "o_orderkey")) %>%
     filter(o_orderstatus == "F") %>%
     group_by(l_orderkey, l_suppkey) %>%
-    summarise(failed_delivery_commit = any(l_receiptdate > l_commitdate)) %>%
+    summarise(failed_delivery_commit = sum(ifelse(l_receiptdate > l_commitdate, 1L, 0L))) %>%
```

Comment:
```suggestion
    summarise(failed_delivery_commit = sum(l_receiptdate > l_commitdate) > 0) %>%
```

Would this work too?

Reply:
I don't think it will because I don't think SQL won't sum logicals like R. Could be missing something but that's what this tells me:
``` r
library(dplyr, warn.conflicts = FALSE)
library(duckdb)
#> Loading required package: DBI

con <- DBI::dbConnect(duckdb::duckdb(), ":memory:")
duckdb_register(con, "iris", iris)

iris %>%
  mutate(log = Species == "setosa") %>%
  summarise(log_sum = sum(log))
#>   log_sum
#> 1      50

tbl(con, "iris") %>%
  mutate(log = Species == "setosa") %>%
  summarise(log_sum = sum(log))
#> Warning: Missing values are always removed in SQL.
#> Use `SUM(x, na.rm = TRUE)` to silence this warning
#> This warning is displayed only once per session.
#> Error: duckdb_prepare_R: Failed to prepare query SELECT SUM("log") AS "log_sum"
#> FROM (
#>   SELECT
#>     "Sepal.Length",
#>     "Sepal.Width",
#>     "Petal.Length",
#>     "Petal.Width",
#>     "Species",
#>     "Species" = 'setosa' AS "log"
#>   FROM "iris"
#> ) "q01"
#> LIMIT 11
#> Error: Binder Error: No function matches the given name and argument types 'sum(BOOLEAN)'. You might need to add explicit type casts.
#>  Candidate functions:
#>  sum(DECIMAL) -> DECIMAL
#>  sum(SMALLINT) -> HUGEINT
#>  sum(INTEGER) -> HUGEINT
#>  sum(BIGINT) -> HUGEINT
#>  sum(HUGEINT) -> HUGEINT
#>  sum(DOUBLE) -> DOUBLE
#> 
#> LINE 1: SELECT SUM("log") AS "log_sum"
#>                ^
```

Reply:
Yeah that look nicer for sure. The issue is that it returns a logical so we will need to add `as.integer` below as well:

```r
  summarise(failed_delivery_commit = sum(as.integer(l_receiptdate > l_commitdate)) > 0) %>%
  group_by(l_orderkey) %>%
  summarise(
    n_supplier = n(),
    num_failed = sum(as.integer(failed_delivery_commit))
  ) %>%
 ```

Reply:
Heh, that's funny — but ok so long as we're still getting the right results at the various scale factors!